db_test.go

```diff
@@ -1303,6 +1303,97 @@ func TestInitializeHeadTimestamp(t *testing.T) {
 	})
 }
 
+func TestDB_LabelNames(t *testing.T) {
+	tests := []struct {
+		sampleLabels1 [][2]string // For checking head and disk.
+		// sampleLabels2 should have atleast 1 label name which is not there in sampleLabels1.
+		sampleLabels2 [][2]string // For check head+disk.
+		exp1          []string    // after adding sampleLabels1.
+		exp2          []string    // after adding sampleLabels1 and sampleLabels2.
+	}{
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name1", ""},
+				[2]string{"name3", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name4", ""},
+				[2]string{"name1", ""},
+			},
+			exp1: []string{"name1", "name2", "name3"},
+			exp2: []string{"name1", "name2", "name3", "name4"},
+		},
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name2", ""},
+				[2]string{"name1", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name6", ""},
+				[2]string{"name0", ""},
+			},
+			exp1: []string{"name1", "name2"},
+			exp2: []string{"name0", "name1", "name2", "name6"},
+		},
+	}
+
+	blockRange := DefaultOptions.BlockRanges[0]
+	// Appends samples into the database.
+	appendSamples := func(db *DB, mint, maxt int64, sampleLabels [][2]string) {
```

Comment:
I see few other test funcs use something like this.
Can we make this func more generic so we can reuse it more tests?

`populateAppender(app Appender,mint, maxt int64, labels) err`

Reply:
I see that most of them are adding samples in a different way. Only maybe 1 or 2 were adding sample in the way it is here (and had offsets of `+j` too). I think making this generalised and taking outside wont add much. 

I can add if you feel that would be better.